daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Node.scala

```diff
@@ -216,7 +213,8 @@ object Node {
       signatories: Set[Party],
       stakeholders: Set[Party],
       key: Option[KeyWithMaintainers[Val]],
-      override val byKey: Boolean // invariant (!byKey || exerciseResult.isDefined)
+      byKey: Option[Boolean] // None if unknown
+      // invariant (!byKey || exerciseResult.isDefined)
```

Comment:
How about enforcing this invariant with a `require`?

Reply:
added.